src/app/plugins/cat.plugin.ts

```diff
@@ -0,0 +1,143 @@
+            const breed = breeds[index].name;
+            const breedID = breeds[index].id;
+            this._breeds.push(breed.toLowerCase())
+            this._breedsID.push(breedID.toLowerCase())
+        }
+    
+        // Doesn't work due to the the way the JSON info is now listed, Would be interested in how to do it with foreach if anyone could tell me.
+        // breeds.forEach((breed) => {
+        //   this._breeds.push(breed.toLowerCase());
+        // });
+      })
+      .catch((err) => console.log(err));
+  }
+
+  public async execute(message: IMessage, args?: string[]){
+
+    var command = this._parseCommand(args || []);
+    
+    if (command.includes("fact")) {
+
+        // fact is not yet supported by the API at this time however this code should work upon its release
+        
+        // command = command.replace("fact", "");
+        // if (command.includes("breeds")) {
+
+        //     message.reply(`Breeds supported: \n\`\`\`\n${this._breeds.join('\n')}\`\`\``);
+        //     return;
+
+        // } else {
+
+        //     var search = "";
+
+        //     if (command != "fact") {
+
+        //         search = command.replace("fact", "");
+
+
+        //         const index = this._breeds.findIndex((curBreed) => {
+        //             return command.includes(curBreed);
+        //         })
+
+        //         breed = `breed_id=${this._breedsID[index]}`;
+        //     }
+
+        //     await this.container.httpService
+        //     .get(`${this._API_URL}facts/${search}`)
+        //     .then((response: IHttpResponse) => {
+
+        //         message.reply('', {
+        //             files: [response.data[0].text],
+        //         });
+        //     })
+
+        //     .catch((err) => console.log(err));
+
+        // }
+
+    } else if (command == "breeds" || command == "breed") {
+        
+        //Simply return the list of suported breeds
+        message.reply(`Breeds supported: \n\`\`\`\n${this._breeds.join('\n')}\`\`\``);
+        return;
+
+    } else if (command == "breads" || command == "bread") {
+
+        // the Easter Egg, my gawd
+        message.reply('', {
+            files: [`https://www.thetoychronicle.com/wp-content/uploads/2016/05/Box-cat-Bread-cat-Rato-workroom-Kim-.jpg`],
+        });
+
+    } else if (command.includes("pic")){
+        // lets the system know that pic has been determined
+
+        command = command.replace("pic", "");
+        var breed = "";
+        var count = 1;
+
+        if (command.includes("bomb")){
+            //processes that the user wants 3 pictures
+            count = 3;
+            command = command.replace("bomb", "");
+        }
+
+        // determines that their was aditional text added on
+        if (command.trim() != "") {
+
+            //finds the breed
+            const index = this._breeds.findIndex((curBreed) => {
+                return command.includes(curBreed);
+            })
```

Comment:
;

Reply:
Not too sure where I am missing a semi-colon here,  although in my current tidying up I have added the semi-colons that I missed. Didn't realise that TypeScript wasn't checking for semi-colons so admittedly I got lazy checking for them.

Reply:
Line 113, after the )